js/edge.js

```diff
@@ -0,0 +1,812 @@
+   *
+   *
+   * @param {?({deviceName: string}|
+    *           {width: number, height: number, pixelRatio: number})} config The
+    *     mobile emulation configuration, or `null` to disable emulation.
+    * @return {!Options} A self reference.
+    */
+  setMobileEmulation(config) {
+    this.options_.mobileEmulation = config;
+    return this;
+  }
+
+  /**
+   * Sets the proxy settings for the new session.
+   * @param {capabilities.ProxyConfig} proxy The proxy configuration to use.
+   * @return {!Options} A self reference.
+   */
+  setProxy(proxy) {
+    this.proxy_ = proxy;
+    return this;
+  }
+
+
+  /**
+   * Converts this options instance to a {@link capabilities.Capabilities}
+   * object.
+   * @param {capabilities.Capabilities=} opt_capabilities The capabilities to
+   *     merge these options into, if any.
+   * @return {!capabilities.Capabilities} The capabilities.
+   */
+  toCapabilities(opt_capabilities) {
+    let caps = opt_capabilities || Capabilities.edge();
+    if (this.getEdgeChromium()) {// chromium edge
+      caps.
+        set(Capability.PROXY, this.proxy_).
+        set(Capability.LOGGING_PREFS, this.logPrefs_).
+        set(OPTIONS_CAPABILITY_KEY, this).
+        set(CAPABILITY_KEY.USE_EDGE_CHROMIUM, true);
+    } else {// legacy edge
+      if (this.proxy_) {
+        caps.set(Capability.PROXY, this.proxy_);
+      }
+      Object.keys(this.options_).forEach(function (key) {
+        caps.set(key, this.options_[key]);
+      }, this);
+      caps.set(CAPABILITY_KEY.USE_EDGE_CHROMIUM, false);
+    }
+    return caps;
+  }
+
+  /**
+   * Converts this instance to its JSON wire protocol representation. Note this
+   * function is an implementation not intended for general use.
+   * @return {!Object}
+   *   The JSON wire protocol representation of this instance.
+   */
+  [Symbols.serialize]() {
+    let json = {};
+    for (let key in this.options_) {
+      if (this.options_[key] != null) {
+        json[key] = this.options_[key];
+      }
+    }
+    // chromium edge
+    if (this.getEdgeChromium() && this.extensions_.length) {
+      json.extensions = this.extensions_.map(function (extension) {
+        if (Buffer.isBuffer(extension)) {
+          return extension.toString('base64');
+        }
+        return io.read(/** @type {string} */(extension))
+          .then(buffer => buffer.toString('base64'));
+      });
+    }
+    return json;
+  }
+}// class Options
+
+/**
+ * Creates {@link remote.DriverService} instances that manage a
+ * MicrosoftEdgeDriver server in a child process.
+ */
+class ServiceBuilder extends remote.DriverService.Builder {
+  /**
+   * @param {string=} opt_exe Path to the server executable to use. If omitted,
+   *   the builder will attempt to locate the MicrosoftEdgeDriver on the current
+   *   PATH.
+   * @throws {Error} If provided executable does not exist, or the
+   *   MicrosoftEdgeDriver cannot be found on the PATH.
+   */
+  constructor(opt_exe) {
+    let exe;
```

Comment:
Please replace this section with:

let exe = opt_exe || io.findInPath(EDGEDRIVER_LEGACY_EXE, true);

This matches the existing edge and chrome behavior. If the user passes an opt_exe that doesn't exist, they will get an error.

Reply:
updated to use function locateSynchronously to find edge driver.